### PR TITLE
[expo-cli][eject] Fix generated orientation in AndroidManifest

### LINKED ISSUE #2429: Eject generates invalid activity orientation in AndroidManifest.xml
## Description

When I set in my `app.json`:
```json
{
  "orientation": "default"
}
```

Then ejecting generates invalid `AndroidManifest.xml` - it just copy-pastes the `default`, which isn't valid manifest value:
```xml
 <activity android:name=".MainActivity" ... android:screenOrientation="default">
      ...
 </activity>
```

## Expected Behavior

`expo eject` should translate the `orientation: default` value to e.g. `unspecified` or some other proper and valid manifest value.

## Observed Behavior

Manifest like shown above is generated - that leads to an error:

```
Android resource linking failed.

AndroidManifest.xml:29:5-34:16: AAPT: error: 'default' is incompatible with attribute screenOrientation (attr) enum [behind=3, fullSensor=10, fullUser=13, landscape=0, locked=14, nosensor=5, portrait=1, reverseLandscape=8, reversePortrait=9, sensor=4, sensorLandscape=6, sensorPortrait=7, unspecified=4294967295, user=2, userLandscape=11, userPortrait=12].
```

## Environment

```
Expo CLI 3.22.3 environment info:
    System:
      OS: macOS 10.15.5
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.17.0 - ~/.volta/tools/image/node/12.17.0/bin/node
      Yarn: 1.22.4 - ~/.volta/tools/image/yarn/1.22.4/bin/yarn
      npm: 6.14.4 - ~/.volta/tools/image/node/12.17.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6514223
      Xcode: 11.5/11E608c - /usr/bin/xcodebuild
    npmPackages:
      expo: ~38.0.0 => 38.0.0 
      react: ~16.11.0 => 16.11.0 
      react-dom: ~16.11.0 => 16.11.0 
      react-native: ~0.62.2 => 0.62.2 
      react-native-web: ~0.11.7 => 0.11.7 
```

## Reproducible Demo

1. `expo init`
2. Edit `app.json` and set `orientation: default`, save
3. `expo eject`
4. Open in Android Studio and build or `yarn android`

### PR DESCRIPTION
**Pull request by barthap**
Fixes #2429 

Tested by just ejecting project and checking if orientation in manifest is correct. Also wrote 2 tests and ran `jest`.

### CODE REVIEW

File: packages/config/src/android/Orientation.ts [hunk 1]
```diff
@@ -16,7 +24,7 @@ export async function setAndroidOrientation(config: ExpoConfig, manifestDocument
   const mainActivity = manifestDocument.manifest.application[0].activity.filter(
     (e: any) => e['$']['android:name'] === '.MainActivity'
   );
-  mainActivity[0]['$'][SCREEN_ORIENTATION_ATTRIBUTE] = orientation;
+  mainActivity[0]['$'][SCREEN_ORIENTATION_ATTRIBUTE] = prepareOrientation(orientation);
```
Comment (wkozyra95) [inline packages/config/src/android/Orientation.ts:27]:
```suggestion
  mainActivity[0]['$'][SCREEN_ORIENTATION_ATTRIBUTE] = orientation !== 'default' ? orientation : 'unspecified';
```
nit pick: prepareOrientation might be a bit misleading and implementation is simple enough to just put inline

File: packages/config/src/android/Orientation.ts [hunk 2]
```diff
@@ -7,6 +7,14 @@ export function getOrientation(config: ExpoConfig) {
   return typeof config.orientation === 'string' ? config.orientation : null;
 }
 
+export function prepareOrientation(orientation: string) {
+  if (orientation === 'default') {
```
Comment (wkozyra95) [inline packages/config/src/android/Orientation.ts:11]: maybe it's a stupid question, but why you are handling 'default' value differently than missing one?

File: packages/config/src/android/Orientation.ts (hunk 1 above)
Comment (barthap) [reply packages/config/src/android/Orientation.ts:27]: You're right - assuming that Expo accepts only 3 values: `default`, `portrait` and `landscape`, and 2 of them are valid manifest values, I can simplify that 😉

File: packages/config/src/android/Orientation.ts (hunk 2 above)
Comment (barthap) [reply packages/config/src/android/Orientation.ts:11]: Just because the missing one is already handled when called this function (line 7)

**Comment by dsokal** [comment]
@barthap could you please update the changelog?

**Comment by bbarthec** [comment]
`CHANGELOG` updated here: https://github.com/expo/expo-cli/commit/274f7bfb83c5bd3f1e577ad5f6f2de886c8f8857

**Pull request merged**